deps/graph-parser/src/logseq/graph_parser/block.cljs

```diff
@@ -146,99 +146,77 @@
+                                                gp-property/editable-linkable-built-in-properties)
                                          (gp-property/hidden-built-in-properties))
                               (keyword k))))
+                   ;; get links ast
                    (map last)
-                   (map (fn [v]
-                          (cond
-                            (and (string? v)
-                                 (not (gp-mldoc/link? format v)))
-                            (let [v (string/trim v)
-                                  result (if (:rich-property-values? user-config)
-                                           (if (gp-util/wrapped-by-quotes? v)
-                                             []
-                                             (text/extract-page-refs-and-tags v))
-                                           (text/split-page-refs-without-brackets v {:un-brackets? false}))]
-                              (if (coll? result)
-                                (map text/page-ref-un-brackets! result)
-                                []))
-
-                            (coll? v)
-                            (map (fn [s]
-                                   (when-not (and (string? v)
-                                                  (gp-mldoc/link? format v))
-                                     (text/page-ref-un-brackets! s))) v)
-
-                            :else
-                            nil)))
-                   (apply concat))
+                   (mapcat (or (:extract-refs-from-property-value-fn user-config)
+                               text/extract-refs-from-mldoc-ast))
+                   ;; comma separated collections
+                   (concat (->> (map second properties)
+                                (filter coll?)
+                                (apply concat))))
         page-refs-from-property-names (get-page-refs-from-property-names properties user-config)]
     (->> (concat page-refs page-refs-from-property-names)
          (remove string/blank?)
          distinct)))
 
 (defn extract-properties
-  [format properties user-config]
+  [properties user-config]
   (when (seq properties)
     (let [properties (seq properties)
-          page-refs (get-page-ref-names-from-properties format properties user-config)
           *invalid-properties (atom #{})
           properties (->> properties
-                          (map (fn [[k v]]
-                                 (let [k (-> (string/lower-case (name k))
+                          (map (fn [[k v mldoc-ast]]
+                                 (let [k (if (or (keyword? k) (symbol? k))
+                                           (subs (str k) 1)
+                                           k)
+                                       k (-> (string/lower-case k)
                                              (string/replace " " "-")
                                              (string/replace "_" "-")
```

Comment:
I think property renaming isn't great for the user overall. Currently the user when writing a property name should be aware of the following:

1. Properties are invalid for certain characters
2. Properties are case insensitive
3. Properties with a `_` get automatically converted to `-`
4. Properties with characters `{}()|"` are stripped

As a user, 1 is explained well with an error message and 2 is understandable because of property pages. Rules 3 and 4 are surprising as what they see and what they edit (and have saved to file) are different. For users and plugin authors that write queries for properties effected by rules 3 and 4, they need to use property names that are different from what's saved to file, at which point it's easier to not use those characters at all

Is rule 3 for backwards compatibility for `background_color`, `created_at` and `last_modified_at`? If so, can't we rename those like we do with custom-id? When I grep for `string/replace.*_.*-` I see a few other places that may need to be updated.
For rule 4, can't we just remove it and let existing invalid property UI explain they are invalid?

Whatever's decided, I'll update the property docs as we don't currently document any of the above

Reply:
Good point!
For rule 4, I've made the changes as you suggested.
I'm not sure about rule 3 though, I agree that it can surprise the user that they see different property keys than what they edit. The benefits of converting `_` to `-` are:
1. When writing queries, users don't have to remember whether a property key is using `_` or `-`. 
2. Backward compatible with the old properties like `background_color` and `created_at`

Reply:
Thanks for one less rule for the user. Good to know rule 3 is for backwards compatibility. Seems reasonable to leave it alone. Users can let us know if they disagree